### PR TITLE
Add importlib-metadata dependency

### LINKED ISSUE #114: Need dependency on importlib-metadata
- [X] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

Sphinx 4.4.0 introduced a dependency on the `importlib-metadata` backport when installed with python < 3.10. (See sphinx-doc/sphinx#10007.) The 4.4.0 conda package does not have that dependency metadata, so installing sphinx 4.4.0 via conda-forge does not install `importlib-metadata` when it should.

Example:

First make an empty project.
```shell
$ conda create -n test -y "sphinx=4.4.0" "python<3.10"
$ conda activate test
$ mkdir test-dir
$ cd test-dir
$ sphinx-quickstart
< accept default options, put in "test" as name, responses don't matter >
```

The way I have found to trigger an error is to use a non-default theme. So I edit the `conf.py` to set `html_theme = 'my_custom_theme'`. It doesn't actually matter what the theme is, I haven't installed it, but it will never be found because of an error.

```shell
$ make html
Running Sphinx v4.4.0
loading pickled environment... done

Exception occurred:
  File "/usr/local/Caskroom/miniconda/base/envs/test/lib/python3.9/site-packages/sphinx/theming.py", line 207, in load_external_theme
    theme_entry_points = entry_points(group='sphinx.html_themes')
TypeError: entry_points() got an unexpected keyword argument 'group'
The full traceback has been saved in /var/folders/rn/37gxvfws6mvcyw4x6k18f_8m0000gp/T/sphinx-err-xhwhv03w.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 2
```

If I then install `importlib-metadata`, I get the (correct, expected) error about my fake theme.
```shell
$ conda install -y importlib-metadata
$ make html
Running Sphinx v4.4.0
loading pickled environment... done

Theme error:
no theme named 'my_custom_theme' found (missing theme.conf?)
make: *** [html] Error 2
```

### PR DESCRIPTION
**Pull request by johnflavin**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Fixes #114

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,26 +9,27 @@ source:
...
-  number: 0
-  noarch: python
+  number: 1
   script: {{ PYTHON }} -m pip install . -vv
   entry_points:
     - sphinx-build = sphinx.cmd.build:main
     - sphinx-quickstart = sphinx.cmd.quickstart:main
     - sphinx-apidoc = sphinx.ext.apidoc:main
     - sphinx-autogen = sphinx.ext.autosummary.generate:main
+  skip: true  # [py<36]
```
Comment (jakirkham) [inline recipe/meta.yaml:19]: As we don't build for anything older than Python 3.6,  this can be dropped

**Review by jakirkham — COMMENTED** [review]
Looks good. Thanks John! 😄

Had one small suggestion below

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml:19]:
```suggestion
```

**Comment by bollwyvl** [comment]
> `/Users/runner/miniforge3/conda-bld/sphinx_1642628795584/work/conda_build.sh: line 4: /Users/runner/miniforge3/conda-bld/sphinx_1642628795584/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/python: Bad CPU type in executable`

That ickiness aside, dropping noarch seems like a heavy price to pay... i've opened https://github.com/conda-forge/sphinx-feedstock/pull/116 as an alternative. it appears they are just using the `entry_points` reader. it won't be strictly _accurate_, but also won't fail, either...

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,26 +9,27 @@ source:
...
     - pip
     - babel
   run:
-    - python >=3.6
+    - python
     - alabaster >=0.7,<0.8
     - babel >=1.3
     - docutils >=0.14,<0.18
     - imagesize
+    - importlib-metadata >=4.4  # [py<310]
```
Comment (bollwyvl) [inline recipe/meta.yaml:32]: note, this _replaces_ `setuptools`, so that can be removed below... c/f https://github.com/sphinx-doc/sphinx/pull/10007

**Comment by johnflavin** [comment]
> dropping noarch seems like a heavy price to pay

I agree. I'm not really a fan of dropping `noarch` and adding a lot more complexity to the build. Which, incidentally, is failing at the moment because I don't have a good grasp of all the build configurations that need to be set. (Clearly the configurations I yoinked from [matplotlib's feedstock](https://github.com/conda-forge/matplotlib-feedstock/blob/master/conda-forge.yml) aren't all working.) Plus there isn't even a build for 3.10, which I would have expected.

I'm going to close this for now in favor of #116.

**Comment by jakirkham** [comment]
Thanks for working on this John! 😄

**Pull request closed without merging**